x/attribute/keeper/keeper.go

```diff
@@ -163,21 +163,103 @@ func (k Keeper) SetAttribute(
 	return nil
 }
 
-// Removes attributes under the given account. The attribute name must resolve to the given owner address.
-func (k Keeper) DeleteAttribute(ctx sdk.Context, acc sdk.AccAddress, name string, owner sdk.AccAddress) error {
+// Updates an attribute under the given account. The attribute name must resolve to the given owner address and value must resolve to an existing attribute.
+func (k Keeper) UpdateAttribute(ctx sdk.Context, originalAttribute types.Attribute, updateAttribute types.Attribute, owner sdk.AccAddress,
+) error {
+	defer telemetry.MeasureSince(time.Now(), types.ModuleName, "keeper_method", "update")
+
+	accountAddress, err := sdk.AccAddressFromBech32(originalAttribute.Address)
+	if err != nil {
+		return err
+	}
+
+	if err = originalAttribute.ValidateBasic(); err != nil {
+		return err
+	}
+
+	if err = updateAttribute.ValidateBasic(); err != nil {
+		return err
+	}
+
+	if updateAttribute.Name != originalAttribute.Name {
+		return fmt.Errorf("update and original names must match %s : %s", originalAttribute.Name, updateAttribute.Name)
+	}
+
+	maxLength := k.GetMaxValueLength(ctx)
+	if int(maxLength) < len(updateAttribute.Value) {
+		return fmt.Errorf("update attribute value length of %v exceeds max length %v", len(updateAttribute.Value), maxLength)
+	}
+
+	normalizedName, err := k.nameKeeper.Normalize(ctx, updateAttribute.Name)
+	if err != nil {
+		return fmt.Errorf("unable to normalize attribute name \"%s\": %w", updateAttribute.Name, err)
+	}
+
+	updateAttribute.Name = normalizedName
+
+	if ownerAcc := k.authKeeper.GetAccount(ctx, owner); ownerAcc == nil {
+		return fmt.Errorf("no account found for owner address \"%s\"", owner.String())
+	}
+
+	if !k.nameKeeper.ResolvesTo(ctx, updateAttribute.Name, owner) {
+		return fmt.Errorf("\"%s\" does not resolve to address \"%s\"", updateAttribute.Name, owner.String())
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	it := sdk.KVStorePrefixIterator(store, types.AccountAttributesNameKeyPrefix(accountAddress, originalAttribute.Name))
```

Comment:
The `originalAttribute.Name` hasn't been normalized, but the `updateAttribute.Name` has. Is it correct to use `originalAttribute.Name` here?